File: mip_reference/dotnet-aspnet/Mip/Identifier.cs
using System.Security.Cryptography;
using System.Text;

namespace Mip;

/// <summary>
/// Generates MIP identifiers per the spec:
/// MD5(UUID + organization_name)
/// </summary>
public static class Identifier
{
    public static string Generate(string organizationName)
    {
        var uuid = Guid.NewGuid().ToString();
        var input = uuid + organizationName;
        var hashBytes = MD5.HashData(Encoding.UTF8.GetBytes(input));
        return Convert.ToHexString(hashBytes).ToLowerInvariant();
    }
}

File: mip_reference/php-laravel/app/Services/Mip/Identifier.php
<?php

namespace App\Services\Mip;

use Ramsey\Uuid\Uuid;

class Identifier
{
    /**
     * Generate a 128-bit MIP identifier using MD5(UUID + organization_name)
     */
    public static function generate(string $organizationName): string
    {
        $uuid = Uuid::uuid4()->toString();
        return md5($uuid . $organizationName);
    }
}
